drivers/netdev2_eth/netdev2_eth.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_netdev2_eth
+ * @{
+ *
+ * @file
+ * @brief       Common code for netdev2 ethernet drivers
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include <assert.h>
+#include <errno.h>
+
+#include "net/netdev2.h"
+#include "net/eui64.h"
+#include "net/ethernet.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static int _get_iid(netdev2_t *netdev, eui64_t *value, size_t max_len)
+{
+    if (max_len < sizeof(eui64_t)) {
+        return -EOVERFLOW;
+    }
+
+    uint8_t addr[ETHERNET_ADDR_LEN];
+    netdev->driver->get(netdev, NETOPT_ADDRESS, addr, ETHERNET_ADDR_LEN);
+    ethernet_get_iid(value, addr);
+
+    return sizeof(eui64_t);
+}
+
+int netdev2_eth_get(netdev2_t *dev, netopt_t opt, void *value, size_t max_len)
+{
+    int res = 0;
+
+    switch (opt) {
+        case NETOPT_DEVICE_TYPE:
```

Comment:
Well in lwIP I use it to set the MTU. GNRC does the same, but it uses 1280 as default, if there is no value given for that by the driver.